Library/Homebrew/utils/pypi.rb

```diff
@@ -93,9 +96,9 @@ def same_package?(other)
     end
 
     # Compare only names so we can use .include? and .uniq on a Package array
-    sig { params(other: Package).returns(T::Boolean) }
+    sig { params(other: T.untyped).returns(T::Boolean) }
```

Comment:
Feels like it might be nicer to keep this typed?

Reply:
I'm not familiar with Sorbet, but part of code I added ended up with `nil` comparison (as `main_package` ended up `nil`). I do think `brew typecheck` still passed, but was thinking this needed at least `T.nilable(Package)` and then just ended up allowing any comparison.

https://github.com/Homebrew/brew/blob/06b7573e0b87eeb67907658d0004548b7c72c0d3/Library/Homebrew/utils/pypi.rb#L292

Still thinking on best way to refactor code to help create a non-pypi `Package` that can be used to perform the `#{package_name}@#{formula.stable.url}` logic that I mentioned in latest comment.